src/server/scripts/Spells/spell_warrior.cpp

```diff
@@ -177,20 +179,33 @@ class spell_warr_charge_effect : public SpellScript
     }
 };
 
-// 167105 - Colossus Smash 7.1.5
+// 167105 - Colossus Smash
 class spell_warr_colossus_smash : public SpellScript
 {
     PrepareSpellScript(spell_warr_colossus_smash);
 
     bool Validate(SpellInfo const* /*spellInfo*/) override
     {
-        return ValidateSpellInfo({ SPELL_WARRIOR_COLOSSUS_SMASH_EFFECT });
+        return ValidateSpellInfo({ SPELL_WARRIOR_COLOSSUS_SMASH_EFFECT, SPELL_WARRIOR_IN_FOR_THE_KILL, SPELL_WARRIOR_IN_FOR_THE_KILL_HASTE });
     }
 
     void HandleOnHit()
     {
-        if (Unit* target = GetHitUnit())
-            GetCaster()->CastSpell(target, SPELL_WARRIOR_COLOSSUS_SMASH_EFFECT, true);
+        Unit* target = GetHitUnit();
+        Unit* caster = GetCaster();
+        caster->CastSpell(target, SPELL_WARRIOR_COLOSSUS_SMASH_EFFECT, true);
+
+        if (caster->HasAura(SPELL_WARRIOR_IN_FOR_THE_KILL))
+            if (SpellInfo const* spellInfo = sSpellMgr->GetSpellInfo(SPELL_WARRIOR_IN_FOR_THE_KILL, DIFFICULTY_NONE))
+            {
+                CastSpellExtraArgs args(TRIGGERED_FULL_MASK);
+                args.AddSpellBP0(spellInfo->GetEffect(EFFECT_0).CalcValue());
```

Comment:
pass `GetTarget()` to `CalcValue()` calls here as it may be modified by other auras (e.g. torghast)

Reply:
Done :)

Reply:
I don't have VS at work :P

Reply:
Actually the correct argument to use here would be the caster

Reply:
it is true, in this case it is the caster who has the aura, done